internal/federationdomain/endpoints/token/token_handler.go

```diff
@@ -79,6 +94,19 @@ func NewHandler(
 	})
 }
 
+func maybeOverrideDefaultAccessTokenLifetime(overrideAccessTokenLifespan timeouts.OverrideLifespan, accessRequest fosite.AccessRequester) {
+	if doOverride, newLifespan := overrideAccessTokenLifespan(accessRequest); doOverride {
+		accessRequest.GetSession().SetExpiresAt(fosite.AccessToken, time.Now().UTC().Add(newLifespan).Round(time.Second))
+	}
+}
+
+func maybeOverrideDefaultIDTokenLifetime(baseCtx context.Context, overrideIDTokenLifespan timeouts.OverrideLifespan, accessRequest fosite.AccessRequester) context.Context {
+	if doOverride, newLifespan := overrideIDTokenLifespan(accessRequest); doOverride {
```

Comment:
I wonder if the returns from this function should be swapped so that the bool is last, similar to `v, ok := claims[claim]`, where `ok == true` lets you know that `v` is useable.

Reply:
Will do.